connexion/decorators/parameter.py

```diff
@@ -87,10 +89,13 @@ def parameter_to_arg(parameters, function):
     def wrapper(*args, **kwargs):
         logger.debug('Function Arguments: %s', arguments)
 
-        try:
-            request_body = flask.request.json
-        except exceptions.BadRequest:
-            request_body = None
+        if 'application/json' in consumes:
```

Comment:
You should use [`is_json_mimetype`](https://github.com/zalando/connexion/blob/master/connexion/utils.py#L137) for this check.